jobs/bootimagebump.Jenkinsfile

```diff
@@ -0,0 +1,145 @@
+                   defaultValue: 'OCPBUGS-48762',
+                   trim: true),
+            text(name: 'JIRA_ISSUES',
+                   description: 'JIRA issues for the bootimage bump',
+                   defaultValue: '',
+                   trim: true),
+            string(name: 'COREOS_ASSEMBLER_IMAGE',
+                   description: 'Override the coreos-assembler image to use',
+                   defaultValue: "quay.io/coreos-assembler/coreos-assembler:rhcos-4.16",
+                   trim: true),
+        ]),
+        buildDiscarder(logRotator(
+            numToKeepStr: '100',
+            artifactNumToKeepStr: '100'
+        )),
+        durabilityHint('PERFORMANCE_OPTIMIZED')
+    ])
+
+    COSA_IMAGE = "quay.io/coreos-assembler/coreos-assembler:rhcos-${params.STREAM}"
+    RHCOS_METADATA_FILE = "data/data/coreos/rhcos.json"
+    PR_BRANCH = "bootimage-bump-${params.BUILD_VERSION}"
+    RELEASE_BRANCH = "release-${params.STREAM.split('-')[0]}"
+
+    cosaPod(serviceAccount: "jenkins",
+            image: params.COREOS_ASSEMBLER_IMAGE,
+            memory: "512Mi", kvm: false,){
+        try {
+            shwrap("""
+                git config --global user.name "CoreOS Bot"
+                git config --global user.email "coreosbot-releng@fedoraproject.org"
+            """)
+
+            stage('Setup workspace') {
+                shwrap("""
+                        git clone --depth=1 --branch main https://github.com/${releng_installer}.git
+                        cd installer
+                        git remote -v
+                        git remote add upstream https://github.com/openshift/installer.git
+                        retries=5
+                        for i in \$(seq 1 \$retries); do
+                            echo "Attempt \$i of \$retries to fetch upstream/${RELEASE_BRANCH}"
+                            git fetch upstream ${RELEASE_BRANCH} && break || sleep 10
+                        done
+                        git checkout -b ${PR_BRANCH} upstream/${RELEASE_BRANCH}
+                        git remote -v
+                """)
+            }
+
+            stage('Bump Bootimage Metadata') {
+                shwrap("""
+                        cd installer
+                        plume cosa2stream \
+                            --target ${RHCOS_METADATA_FILE} \
+                            --distro rhcos \
+                            --no-signatures \
+                            --name ${params.STREAM} \
+                            --url https://rhcos.mirror.openshift.com/art/storage/prod/streams \
+                            x86_64=${params.BUILD_VERSION} \
+                            aarch64=${params.BUILD_VERSION} \
+                            s390x=${params.BUILD_VERSION} \
+                            ppc64le=${params.BUILD_VERSION}
+                """)
+            }
+
+            stage('Create Pull Request') {
+                //if (shwrapCapture("git diff --exit-code") != 0){
```

Comment:
Using git diff --exit-code seem to throw an issue here. 

13:55:10  + git diff --exit-code
13:55:10  warning: Not a git repository. Use --no-index to compare two paths outside a working tree
13:55:10  usage: git diff --no-index [<options>] <path> <path>

Reply:
This isn't in the `installer` directory so it is failing there.  You can try `git -C installer diff --exit-code`

Reply:
With the git -C installer diff --exit-code , it seems to print out the output of git diff and then just exit the code with a failure, not quite evident why it is exiting after that check.